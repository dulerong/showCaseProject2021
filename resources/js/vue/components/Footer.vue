<template lang="pug">
v-footer(padless absolute app)
  v-card(width="100%")
    v-card-text
      v-row(no-gutters justify="center")
        v-btn(v-for="(item, i) in $_footer_navigation" :key="i" icon class="mx-4" :to="{ name: item.name, params: { lang: $i18n.locale } }" exact)
          ToolTip(:message="$t(`footer.tooltip.${item.name}`)")
            v-icon {{ item.icon }}
    v-divider
    v-card-text
      v-row(no-gutters justify="center")
        h1 {{ $_dateToday.format('YYYY') }} - Codey Du
</template>

<script>
export default {
  components: {
    ToolTip: require('~components/ToolTip').default,
  },
}
</script>