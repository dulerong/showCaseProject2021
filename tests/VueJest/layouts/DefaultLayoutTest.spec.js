import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vuetify from 'vuetify'

import defaultLayout from '~layouts/default'
import TopBar from '~components/crm/TopBar'
import SideBar from '~components/crm/SideBar'
import Footer from '~components/crm/Footer'

describe('App Component Tests', () => {
  const localVue = createLocalVue()
  let vuetify

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  const shallowMountFunction = options => {
    return shallowMount(defaultLayout, {
      localVue,
      vuetify,
      ...options,
    })
  }

  it('App renders properly', () => {
    const wrapper = shallowMountFunction()

    expect(wrapper.exists()).toBe(true)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('Component exists', () => {
    const wrapper = shallowMountFunction()

    const componentTopBar = wrapper.findComponent(TopBar)
    expect(componentTopBar.exists()).toBe(true)

    const componentSideBar = wrapper.findComponent(SideBar)
    expect(componentSideBar.exists()).toBe(true)

    const componentFooter = wrapper.findComponent(Footer)
    expect(componentFooter.exists()).toBe(true)
  })
})
