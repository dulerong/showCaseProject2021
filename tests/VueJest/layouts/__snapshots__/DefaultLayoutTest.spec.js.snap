// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Tests App renders properly 1`] = `
<v-app-stub id="app">
  <appbar-stub toggle="[Function]" toggleside="[Function]"></appbar-stub>
  <navigationdrawer-stub drawer="[object Object]"></navigationdrawer-stub>
  <settingdrawer-stub sidedrawer="[object Object]"></settingdrawer-stub>
  <v-main-stub tag="main">
    <div class="pa-4 mx-auto" style="max-width: 1280px; height: 100%;"></div>
  </v-main-stub>
  <snackbar-stub></snackbar-stub>
  <footer-stub></footer-stub>
</v-app-stub>
`;
